go/cmd/dolt/commands/commit.go

```diff
@@ -150,7 +157,7 @@ func (cmd CommitCmd) Exec(ctx context.Context, commandStr string, args []string,
 	pendingCommit, err := actions.GetCommitStaged(ctx, roots, ws.MergeActive(), mergeParentCommits, dEnv.DbData(), actions.CommitStagedProps{
 		Message:    msg,
 		Date:       t,
-		AllowEmpty: apr.Contains(cli.AllowEmptyFlag),
+		AllowEmpty: apr.Contains(cli.AllowEmptyFlag) || apr.Contains(cli.AmendFlag),
```

Comment:
What was the reason this was needed? To account for amending a commit that was already an empty commit?

Reply:
This is to account for commit amends where only the commit message is changed. For these cases, the new "commit" after the soft reset will not have any staged tables. This change is hence to prevent `GetCommitStaged` from returning an error.